### ISSUE TITLE
Cache loaded schema / Improve performance

### ISSUE THREAD

**Issue by piglovesyou** [piglovesyou/graphql-let#24]
- [x] Investigate bottleneck
    - [x] for the command<br>→#35 improves it
    - [x] for the loader<br>→Caching schema solves heap error
- [x] Cache loaded schema

**Comment by ALITHEALI** [comment]
Hello,
Is there any update on the performance? I have an app with 200+ mutations and with the current performance it will take me 30 mins to generate the types. Let me know if I can help with anything.
Thanks in advance.

**Comment by piglovesyou** [comment]
Hi @ALITHEALI , wow, thanks for reporting. >30 mins. I will:

* Try with 200+ documents to reproduce
    * Compare to the graphql-codegen command
* Release the fix of the known issue soon - Cache loaded schema
* study more if worker-lizing the proccesses would be our choice

**Comment by ALITHEALI** [comment]
Thanks a lot. I think I can take a look tomorrow after work to see if I can help in anyway shape or form I can because this project has single-handedly solved numerous problems we had with normal plain GraphQL Code Generator since yesterday that we tried it. Hope to see this project reach to the popularity it deserves and thank you for creating it.

**Comment by piglovesyou** [comment]
@ALITHEALI My pleasure to hear that. This must be off-topic though, I'm curious about what the "numerous problems" were. If possible, I'd like to explain them in the README.md.

**Comment by piglovesyou** [comment]
@ALITHEALI [The v0.5.2](https://www.npmjs.com/package/graphql-let/v/0.5.2) or later must solve the slowness of the graphql-let command. Glad if you take a look.

**Comment by ALITHEALI** [comment]
Hello! So sorry about the late reply I was buried under work. Of course, I will list the issues below: 

1.  One of the major issues was that the separation of file exports are not fully functional in graphql-codegen. Of course you can use their native option to generate code per document but in our experience it needed a lot work since the generated code was not complete and sometimes lacked some Hooks. 
2. NO SEPARATE TYPING! This was so hard that we had to copy and paste our typings into our .d.ts files in order to use the typing with Golang since we have a tool to generate code for Golang for our other microservices.
3. This third one is my favorite because previously we had to import the .tsx file in order to use the "useFoo" hooks and HOCs in our files but since the last time I followed up on this issue we have generated a tool which can merge all the .gql files before codegen and make it 1 file and we import all the server functions we need from that file. this means, less code, less drama and less over dependence on generated code (at least on the surface level) and easier to find functions in our team since that file literally contains our schema, hooks and HOCs 🔥and this is one of the main reasons I said this package can change the game, because every single thing that you want from server is in **1 FILE**  and can compile with no problem!
4. Another issue we had with normal plain gql-codegen was that it takes way more time to load the typescript types for some reason and since I haven't checked this repo's code I cannot see why it is loading waaay faster but soon I will dive into it. 

TLDR; Typescript def files generation, functional file exports, no "code-bumps" , **1 File** for all server stuff, faster build time and faster load time and faster IDE types loader. 

About the v0.5.2, I have checked it and right off the bat, it's very much faster than the previous command, soon I will dive into this repo's code to see if I can help.

Thanks again.

**Comment by piglovesyou** [comment]
Thank you for the kind words, it's happy to hear that. About the faster IDE assists, it may relate to the smaller .d.ts, since graphql-codegen usually binds many .gql files into one/less number of files. And for the others, I guess generating .d.ts instead of .tsx might help your project.

I'm relieved to hear v0.5.2 shorten the build time. Please make comments if you notice something on this issue/a new issue, anytime.
